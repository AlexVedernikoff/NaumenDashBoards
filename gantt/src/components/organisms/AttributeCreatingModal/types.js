// @flow
import type {ComputedAttr} from 'store/widgets/data/types';
import {CONTROL_TYPES} from './constants';
import type {Node, Tree} from 'components/molecules/TreeSelect/types';

export type ControlType = $Keys<typeof CONTROL_TYPES>;

export type Control = {
	name: string,
	type: ControlType,
	value: null
};

export type Props = {
	onClose: () => any,
	onFetch: (node: Node) => void,
	onRemove?: (attribute: ComputedAttr) => void,
	onSubmit: (attribute: ComputedAttr) => void,
	options: Tree,
	value: ComputedAttr | null
};

export type State = {
	controls: Array<Control>,
	info: string,
	showFormulaError: boolean,
	showLegacyFormatInfo: boolean,
	showRemoveInfo: boolean,
	templates: Array<Control>,
	title: string
};
