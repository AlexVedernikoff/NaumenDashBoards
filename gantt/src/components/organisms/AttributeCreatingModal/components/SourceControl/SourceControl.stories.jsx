import {action} from '@storybook/addon-actions';
import React from 'react';
import {SourceControl} from './SourceControl';
import {useArgs} from '@storybook/client-api';

export default {
	component: SourceControl,
	title: 'Organisms/AttributeCreatingModal/components/SourceControl'
};

const Template = args => {
	const [{value}, updateArgs] = useArgs();
	const onSelect = (index, name, value, type) => {
		action('onSelect')(index, name, value, type);
		updateArgs({value});
	};

	return (
		<div style={{width: 60}}>
			<SourceControl
				{...args}
				onClick={action('onClick')}
				onFetch={action('onFetch')}
				onSelect={onSelect} value={value}
			/>
		</div>
	);
};

export const Component = Template.bind({});

Component.args = {
	index: 0,
	options: {
		attr1: {
			children: null,
			id: 'attr1',
			parent: 'dataKey1',
			value: {
				code: 'attr1',
				title: 'attr1'
			}
		},
		attr2: {
			children: null,
			id: 'attr2',
			parent: 'dataKey1',
			value: {
				code: 'attr2',
				title: 'attr2'
			}
		},
		dataKey1: {
			children: ['attr1', 'attr2'],
			id: 'dataKey1',
			value: {
				label: 'Источник 1',
				value: 'source1'
			}
		}
	}
};
