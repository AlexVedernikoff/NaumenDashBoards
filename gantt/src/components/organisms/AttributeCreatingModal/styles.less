@import '/styles/variables';

:local {
	.container {
		min-height: 300px;
		padding: 24px;
	}

	.controlsContainer {
		display: flex;
		flex-flow: row wrap;
	}

	.controlContainer {
		margin-right: 5px;
		margin-bottom: 5px;
		position: relative;

		&:hover {
			.removeIcon {
				display: block;
			};
		}
	}

	.removeIcon {
		border-radius: @border-radius;
		color: @dimgrey;
		cursor: pointer;
		display: none;
		position: absolute;
		right: 0;
		top: 0;

		& :hover {
			background-color: rgba(0, 0, 0, 0.2);
		}
	}

	.nameContainer {
		margin-bottom: 24px;
		position: relative;
		width: 100%;

		&:hover {
			.nameClearIcon {
				display: block;
			}
		}
	}

	.nameLabel {
		color: @darkgrey;
		font-size: 13px;
		margin-bottom: 5px;
	}

	.nameInput {
		border: 1px solid @grey;
		border-radius: @border-radius;
		box-sizing: border-box;
		padding: 8px 12px;
		width: 100%;
	}

	.nameClearIcon {
		color: @dimgrey;
		cursor: pointer;
		display: none;
		height: 14px;
		position: absolute;
		right: 8px;
		top: 30px;
		width: 14px;
	}

	.footer {
		display: flex;
	}

	.footerRemoveButton {
		margin-left: auto;
	}

	.saveButton {
		margin-right: 10px;
	}

	.infoPanel {
		margin-bottom: 24px;
	}
}
