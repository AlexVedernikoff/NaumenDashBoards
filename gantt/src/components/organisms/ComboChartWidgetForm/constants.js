// @flow
import {array, baseSchema, mixed, object} from 'containers/DiagramWidgetForm/schema';
import {COMBO_TYPES} from 'store/widgets/data/constants';
import {DIAGRAM_FIELDS} from 'WidgetFormPanel/constants';

const schema = object({
	...baseSchema,
	data: array().of(object({
		breakdown: mixed().when(DIAGRAM_FIELDS.type, {
			else: mixed().requiredByCompute(array().conditionalBreakdown()),
			is: type => type === COMBO_TYPES.COLUMN_STACKED,
			then: array().breakdown()
		}),
		indicators: mixed().requiredByCompute(array().indicators()),
		parameters: array().parameters(),
		source: mixed().source(),
		top: object().topSettings()
	})),
	sources: mixed().minSourceNumbers()
});

export {
	schema
};
