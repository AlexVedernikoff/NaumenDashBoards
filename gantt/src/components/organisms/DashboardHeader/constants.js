import {FILE_VARIANTS} from 'utils/export';

const EXPORT_LIST = [
	{
		label: FILE_VARIANTS.PDF,
		value: FILE_VARIANTS.PDF
	},
	{
		label: FILE_VARIANTS.PNG,
		value: FILE_VARIANTS.PNG
	}
];

const DASHBOARD_HEADER_HEIGHT = 52;

export {
	DASHBOARD_HEADER_HEIGHT,
	EXPORT_LIST
};
