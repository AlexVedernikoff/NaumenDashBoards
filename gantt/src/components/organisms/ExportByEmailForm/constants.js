// @flow
import {FILE_VARIANTS} from 'utils/export';

const FORMATS = [
	{
		label: 'PDF',
		value: FILE_VARIANTS.PDF
	},
	{
		label: 'PNG',
		value: FILE_VARIANTS.PNG
	}
];

export {
	FORMATS
};
