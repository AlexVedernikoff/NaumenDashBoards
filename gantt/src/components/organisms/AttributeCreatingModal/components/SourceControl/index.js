// @flow
import SourceControl from './SourceControl';
export default SourceControl;
