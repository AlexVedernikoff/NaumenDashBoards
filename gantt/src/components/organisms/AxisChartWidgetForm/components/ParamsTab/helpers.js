// @flow
import type {ChartSorting} from 'store/widgets/data/types';
import {GROUP_WAYS} from 'store/widgets/constants';
import type {Parameter} from 'store/widgetForms/types';
import {SORTING_VALUES} from 'store/widgets/data/constants';

const getSortValue = (parameters: Array<Parameter>, sorting: ChartSorting) => {
	const {group} = parameters[0];
	const {DEFAULT, INDICATOR} = SORTING_VALUES;
	let {value} = sorting;

	if (group.way === GROUP_WAYS.CUSTOM && sorting.value !== DEFAULT) {
		value = DEFAULT;
	} else if (sorting.value === DEFAULT) {
		value = INDICATOR;
	}

	return value;
};

export {
	getSortValue
};
