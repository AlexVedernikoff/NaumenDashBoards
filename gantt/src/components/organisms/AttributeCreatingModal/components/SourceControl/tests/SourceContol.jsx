import {CONTROL_TYPES} from 'components/organisms/AttributeCreatingModal/constants';
import {getAttribute, getProps, getSource} from './helpers';
import {getNode} from 'components/molecules/TreeSelect/components/Node/tests/helpers';
import {mount, shallow} from 'enzyme';
import React from 'react';
import SourceControl from 'components/organisms/AttributeCreatingModal/components/SourceControl';

describe('AttributeCreatingModal SourceControl', () => {
	it('renders when property "value" is not set to "null"', () => {
		const wrapper = mount(<SourceControl {...getProps()} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('renders when property "value" is set to "null"', () => {
		const wrapper = mount(<SourceControl {...getProps()} value={null} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('handles onAddConstant properly and calls provided callback', () => {
		const mockOnAddConstant = jest.fn();
		const props = getProps();
		const wrapper = shallow(<SourceControl {...props} onAddConstant={mockOnAddConstant} />);

		wrapper.instance().handleAddConstant();

		expect(mockOnAddConstant.mock.calls.length).toEqual(1);
		expect(mockOnAddConstant.mock.calls[0][0]).toEqual(props.index);
		expect(mockOnAddConstant.mock.calls[0][1]).toEqual(props.name);
	});

	it('handles onSelect properly and calls provided callback', () => {
		const mockOnSelect = jest.fn();
		const props = getProps();
		const dataKey = 'dataKey';
		const node = getNode({
			parent: dataKey,
			value: getAttribute()
		});
		const options = {
			[dataKey]: getNode({value: getSource()})
		};
		const wrapper = shallow(<SourceControl {...props} onSelect={mockOnSelect} options={options} />);

		wrapper.instance().handleSelect({value: node});

		expect(mockOnSelect.mock.calls.length).toEqual(1);
		expect(mockOnSelect.mock.calls[0][0]).toEqual(props.index);
		expect(mockOnSelect.mock.calls[0][1]).toEqual(props.name);
		expect(mockOnSelect.mock.calls[0][2]).toMatchObject(props.value);
		expect(mockOnSelect.mock.calls[0][3]).toEqual(CONTROL_TYPES.SOURCE);
	});

	it('method "getNodeLabel" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const sourceId = 'dataKey';
		const sourceNode = getNode({id: sourceId, value: getSource()});
		const attributeNode = getNode({parent: sourceId, value: getAttribute()});

		expect(component.getNodeLabel(sourceNode)).toEqual(sourceNode.value.label);
		expect(component.getNodeLabel(attributeNode)).toEqual(attributeNode.value.title);
	});

	it('method "getNodeValue" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const sourceId = 'dataKey';
		const sourceNode = getNode({id: sourceId, value: getSource()});
		const attributeNode = getNode({parent: sourceId, value: getAttribute()});

		expect(component.getNodeValue(sourceNode)).toEqual(sourceNode.id);
		expect(component.getNodeValue(attributeNode)).toEqual(attributeNode.value.code);
	});

	it('method "getOptionLabel" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const attribute = getAttribute();

		expect(component.getOptionLabel(attribute)).toEqual(attribute.title);
		expect(component.getOptionLabel(null)).toEqual('');
	});

	it('method "getOptionValue" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const attribute = getAttribute();

		expect(component.getOptionValue(attribute)).toEqual(attribute.code);
		expect(component.getOptionLabel(null)).toEqual('');
	});

	it('method "isDisabled" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const sourceId = 'dataKey';
		const sourceNode = getNode({id: sourceId, value: getSource()});
		const attributeNode = getNode({parent: sourceId, value: getAttribute()});

		expect(component.isDisabled(sourceNode)).toEqual(true);
		expect(component.isDisabled(attributeNode)).toEqual(false);
	});

	it('method "isSelected" works correctly', () => {
		const component = shallow(<SourceControl {...getProps()} />).instance();
		const sourceId = 'dataKey';
		const sourceNode = getNode({id: sourceId, value: getSource()});
		const attributeNode = getNode({parent: sourceId, value: getAttribute()});

		expect(component.isSelected(sourceNode)).toEqual(false);
		expect(component.isSelected(attributeNode)).toEqual(true);
	});
})
