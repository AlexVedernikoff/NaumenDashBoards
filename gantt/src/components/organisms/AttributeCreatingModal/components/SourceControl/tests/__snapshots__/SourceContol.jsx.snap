// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeCreatingModal SourceControl renders when property "value" is not set to "null" 1`] = `
<SourceControl
  index={0}
  name="name"
  onClick={[Function]}
  onFetch={[Function]}
  onSelect={[Function]}
  options={Object {}}
  value={
    Object {
      "aggregation": "COUNT_CNT",
      "attribute": Object {
        "code": "code",
        "title": "attribute",
      },
      "dataKey": "dataKey",
      "source": Object {
        "label": "source",
        "value": "classFqn",
      },
    }
  }
>
  <TreeSelect
    className="control"
    components={
      Object {
        "MenuContainer": [Function],
        "Tree": [Function],
        "ValueContainer": [Function],
      }
    }
    getNodeLabel={[Function]}
    getNodeValue={[Function]}
    getOptionLabel={[Function]}
    getOptionValue={[Function]}
    isDisabled={[Function]}
    isSelected={[Function]}
    loading={false}
    multiple={false}
    name=""
    onFetch={[Function]}
    onSelect={[Function]}
    options={Object {}}
    placeholder="Выберите значение"
    removable={false}
    showMore={false}
    value={
      Object {
        "code": "code",
        "title": "attribute",
      }
    }
    values={Array []}
  >
    <div
      className="select control"
    >
      <Component
        className="valueContainer"
        onClick={[Function]}
      >
        <div
          className="valueContainer"
        >
          <AttributeAggregationField
            attribute={
              Object {
                "code": "code",
                "title": "attribute",
              }
            }
            onSelect={[Function]}
            renderValue={[Function]}
            tip="Агрегация"
            usesNotApplicableAggregation={false}
            value="COUNT_CNT"
          >
            <MiniSelect
              className=""
              isDisabled={false}
              onSelect={[Function]}
              options={
                Array [
                  Object {
                    "label": "CNT",
                    "value": "COUNT_CNT",
                  },
                  Object {
                    "label": "%",
                    "value": "PERCENT",
                  },
                ]
              }
              renderValue={[Function]}
              showCaret={true}
              tip="Агрегация"
              value="COUNT_CNT"
            >
              <OutsideClickDetector
                onClickOutside={[Function]}
              >
                <div
                  className="select "
                >
                  <div
                    className="valueContainer attributeField"
                    onClick={[Function]}
                  >
                    <div
                      className="label"
                    >
                      CNT
                    </div>
                    <Icon
                      className=""
                      height={16}
                      name="CHEVRON"
                      title=""
                      viewBox="0 0 16 16"
                      width={16}
                    >
                      <svg
                        className="icon"
                        fill="currentColor"
                        height={16}
                        viewBox="0 0 16 16"
                        width={16}
                      >
                        <use
                          className="use"
                          xlinkHref="#CHEVRON"
                        />
                      </svg>
                    </Icon>
                  </div>
                </div>
              </OutsideClickDetector>
            </MiniSelect>
          </AttributeAggregationField>
          <div
            className="labelContainer"
            onClick={[Function]}
          >
            <div
              className="sourceLabel"
            >
              source
            </div>
            <div
              className="attributeLabel"
            >
              attribute
            </div>
          </div>
        </div>
      </Component>
    </div>
  </TreeSelect>
</SourceControl>
`;

exports[`AttributeCreatingModal SourceControl renders when property "value" is set to "null" 1`] = `
<SourceControl
  index={0}
  name="name"
  onClick={[Function]}
  onFetch={[Function]}
  onSelect={[Function]}
  options={Object {}}
  value={null}
>
  <TreeSelect
    className="control"
    components={
      Object {
        "MenuContainer": [Function],
        "Tree": [Function],
        "ValueContainer": [Function],
      }
    }
    getNodeLabel={[Function]}
    getNodeValue={[Function]}
    getOptionLabel={[Function]}
    getOptionValue={[Function]}
    isDisabled={[Function]}
    isSelected={[Function]}
    loading={false}
    multiple={false}
    name=""
    onFetch={[Function]}
    onSelect={[Function]}
    options={Object {}}
    placeholder="Выберите значение"
    removable={false}
    showMore={false}
    value={null}
    values={Array []}
  >
    <div
      className="select control"
    >
      <Component
        className="valueContainer"
        onClick={[Function]}
      >
        <div
          className="valueContainer"
        >
          <div
            className="placeholder"
            onClick={[Function]}
          >
            <Icon
              className=""
              height={16}
              name="ELLIPSIS"
              title=""
              viewBox="0 0 16 16"
              width={16}
            >
              <svg
                className="icon"
                fill="currentColor"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <use
                  className="use"
                  xlinkHref="#ELLIPSIS"
                />
              </svg>
            </Icon>
          </div>
        </div>
      </Component>
    </div>
  </TreeSelect>
</SourceControl>
`;
